### PR TITLE
[WIP] Redesign transaction flow

### LINKED ISSUE #2008: Recently used destination addresses aren't displayed in History tab
My friend sent me his address as a WhatsApp message, therefore I couldn't copy and paste it (I had to type it manually in address field of Send tab).

After 10 minutes, I wanted to send some XMR again to my friend, but I couldn't find his address anywhere on the GUI wallet. Unfortunately the History tab didn't display the destination address of my recent transaction (it displayed "Unkown Recipient"). So I had to type his address all over again.

So my suggestion is to display recently used destination addresses in the History Tab.

### LINKED ISSUE #2467: Transaction confirmation improvements
image

1) Simple mode shouldn't display `Ring size`, `Number of transactions` and `Spending address index`.

2) There should be an additional field `From:`, with the following format:
- From: "_Wallet_" / "_Account_"
- `From: John's wallet / Savings`

3) `Address:` should be renamed into `To:`

4) `Amount:` and `Fee:` should display XMR

5) Amount converted in fiat (USD) should also appear
- in a second line: `Amount (USD):` and `Amount (EUR):`
- or in the same line: `Amount: 1.0 XMR (54 USD)`

6) `Fee:` shouldn't display extra zeros at the end
- Currently: `Fee: 0.0000248200`
- Expected: `Fee: 0.00002482`

7) `Ok` button should be renamed into `Confirm transaction`

### LINKED ISSUE #2703: Transactions: Search field doesn't filter transactions when there is a pending transaction
GUI v0.15.0.2 (Qt 5.9.7)

To reproduce:
1) Send a transaction
2) Go to Transactions page
3) Use search filter. Filter doesn't work while transaction status is Pending
5) When transaction has 1 confirmation, filter starts working again

### PR DESCRIPTION
**Pull request by rating89us**
Closes #2467

**To do:**
- Test Ledger integration
- Test sweep unmixable feature
- Test OpenAlias
- #753

**Changes:**

New TxConfirmationDialog component:
- From field: wallet name ("Ledger", "Trezor" or "My wallet") + account # + account label
- To field: displays icon + address book name (if address is in address book);
- Added fiat conversion to amount and fee
- Removed extra zeros from the end of amount and fee
- Keyboard can be used to cancel (ESC key) or to confirm dialogs (enter/return keys)
- When the dialog is waiting for user action, a blinking text message will appear (see Trezor example below)

Modified StandardButton component:
- Added secondary button: a grey button for not desired actions

Modified PasswordDialog component:
- Ok button text changes to "Send transaction" when sending tx
- Ok button text changes to "Save transaction file" when saving tx file in View only wallet

New SuccessfulTxDialog component:
- Added green circle checkmark animation
- View progress button: opens History page with transaction ID filled in search input
- Open folder button (view-only wallet): opens the folder where the transaction file was saved

History page:
- Destination address of pending (0 confirmation) transactions is now being displayed (fixes #2008)
- Search filter now works when there is a pending transaction (fixes #2703)
- Auto collapse when there is a single result

Example 1: Local wallet file paying to address included in Address Book
txflow3

Example 2: Trezor paying to address not included in Address Book
trezornew3

### CODE REVIEW

File: components/TxConfirmationDialog.qml [hunk 1]
```diff
@@ -0,0 +1,482 @@
...
+        id: mainLayout
+        spacing: 10
+        anchors.fill: parent
+        anchors.margins: 25
+
+        RowLayout {
+            id: column
+            Layout.topMargin: 10
+            Layout.fillWidth: true
+            anchors.top: parent.top
```
Comment (selsta) [inline components/TxConfirmationDialog.qml]: Quick note here. Mixing Layouts and anchors is undefined behaviour.

See here a list of warnings:

```
2020-02-16 20:23:54.926	W qrc:/components/TxConfirmationDialog.qml:163:9: QML RowLayout: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
2020-02-16 20:23:54.927	W qrc:/components/TxConfirmationDialog.qml:189:9: QML QQuickText: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
2020-02-16 20:23:54.927	W qrc:/components/TxConfirmationDialog.qml:200:9: QML QQuickText: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
2020-02-16 20:23:54.928	W qrc:/components/TxConfirmationDialog.qml:211:9: QML BusyIndicator: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
2020-02-16 20:23:54.928	W qrc:/components/TxConfirmationDialog.qml:219:9: QML QQuickText: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
2020-02-16 20:23:54.928	W qrc:/components/TxConfirmationDialog.qml:230:9: QML GridLayout: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
2020-02-16 20:23:54.928	W qrc:/components/TxConfirmationDialog.qml:364:9: QML RowLayout: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
2020-02-16 20:23:54.928	W qrc:/components/TxConfirmationDialog.qml:398:9: QML RowLayout: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
2020-02-16 20:23:54.928	W qrc:/components/TxConfirmationDialog.qml:163:9: QML RowLayout: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
2020-02-16 20:23:54.928	W qrc:/components/TxConfirmationDialog.qml:189:9: QML QQuickText: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
2020-02-16 20:23:54.928	W qrc:/components/TxConfirmationDialog.qml:200:9: QML QQuickText: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
2020-02-16 20:23:54.928	W qrc:/components/TxConfirmationDialog.qml:211:9: QML BusyIndicator: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
2020-02-16 20:23:54.928	W qrc:/components/TxConfirmationDialog.qml:219:9: QML QQuickText: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
2020-02-16 20:23:54.928	W qrc:/components/TxConfirmationDialog.qml:230:9: QML GridLayout: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
2020-02-16 20:23:54.928	W qrc:/components/TxConfirmationDialog.qml:364:9: QML RowLayout: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
2020-02-16 20:23:54.928	W qrc:/components/TxConfirmationDialog.qml:398:9: QML RowLayout: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
2020-02-16 20:23:54.928	W qrc:/components/TxConfirmationDialog.qml:163:9: QML RowLayout: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
2020-02-16 20:23:54.929	W qrc:/components/TxConfirmationDialog.qml:189:9: QML QQuickText: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
2020-02-16 20:23:54.929	W qrc:/components/TxConfirmationDialog.qml:200:9: QML QQuickText: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
2020-02-16 20:23:54.929	W qrc:/components/TxConfirmationDialog.qml:211:9: QML BusyIndicator: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
2020-02-16 20:23:54.929	W qrc:/components/TxConfirmationDialog.qml:219:9: QML QQuickText: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
2020-02-16 20:23:54.929	W qrc:/components/TxConfirmationDialog.qml:230:9: QML GridLayout: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
2020-02-16 20:23:54.929	W qrc:/components/TxConfirmationDialog.qml:364:9: QML RowLayout: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
2020-02-16 20:23:54.929	W qrc:/components/TxConfirmationDialog.qml:398:9: QML RowLayout: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
2020-02-16 20:23:54.929	W qrc:/components/TxConfirmationDialog.qml:163:9: QML RowLayout: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
2020-02-16 20:23:54.929	W qrc:/components/TxConfirmationDialog.qml:189:9: QML QQuickText: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
2020-02-16 20:23:54.929	W qrc:/components/TxConfirmationDialog.qml:200:9: QML QQuickText: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
2020-02-16 20:23:54.929	W qrc:/components/TxConfirmationDialog.qml:211:9: QML BusyIndicator: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
2020-02-16 20:23:54.929	W qrc:/components/TxConfirmationDialog.qml:219:9: QML QQuickText: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
2020-02-16 20:23:54.929	W qrc:/components/TxConfirmationDialog.qml:230:9: QML GridLayout: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
2020-02-16 20:23:54.929	W qrc:/components/TxConfirmationDialog.qml:364:9: QML RowLayout: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
2020-02-16 20:23:54.929	W qrc:/components/TxConfirmationDialog.qml:398:9: QML RowLayout: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
2020-02-16 20:23:54.929	W qrc:/components/TxConfirmationDialog.qml:163:9: QML RowLayout: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
2020-02-16 20:23:54.929	W qrc:/components/TxConfirmationDialog.qml:189:9: QML QQuickText: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
2020-02-16 20:23:54.929	W qrc:/components/TxConfirmationDialog.qml:200:9: QML QQuickText: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
2020-02-16 20:23:54.929	W qrc:/components/TxConfirmationDialog.qml:211:9: QML BusyIndicator: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
2020-02-16 20:23:54.929	W qrc:/components/TxConfirmationDialog.qml:219:9: QML QQuickText: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
2020-02-16 20:23:54.929	W qrc:/components/TxConfirmationDialog.qml:230:9: QML GridLayout: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
2020-02-16 20:23:54.929	W qrc:/components/TxConfirmationDialog.qml:364:9: QML RowLayout: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
2020-02-16 20:23:54.929	W qrc:/components/TxConfirmationDialog.qml:398:9: QML RowLayout: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
2020-02-16 20:23:54.929	W qrc:/components/TxConfirmationDialog.qml:163:9: QML RowLayout: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
2020-02-16 20:23:54.929	W qrc:/components/TxConfirmationDialog.qml:189:9: QML QQuickText: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
2020-02-16 20:23:54.929	W qrc:/components/TxConfirmationDialog.qml:200:9: QML QQuickText: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
2020-02-16 20:23:54.930	W qrc:/components/TxConfirmationDialog.qml:211:9: QML BusyIndicator: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
2020-02-16 20:23:54.930	W qrc:/components/TxConfirmationDialog.qml:219:9: QML QQuickText: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
2020-02-16 20:23:54.930	W qrc:/components/TxConfirmationDialog.qml:230:9: QML GridLayout: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
2020-02-16 20:23:54.930	W qrc:/components/TxConfirmationDialog.qml:364:9: QML RowLayout: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
2020-02-16 20:23:54.930	W qrc:/components/TxConfirmationDialog.qml:398:9: QML RowLayout: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
2020-02-16 20:23:54.931	W qrc:/components/SuccessfulTxDialog.qml:109:9: QML ColumnLayout: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
2020-02-16 20:23:54.931	W qrc:/components/SuccessfulTxDialog.qml:167:9: QML RowLayout: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
2020-02-16 20:23:54.931	W qrc:/components/SuccessfulTxDialog.qml:109:9: QML ColumnLayout: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
2020-02-16 20:23:54.931	W qrc:/components/SuccessfulTxDialog.qml:167:9: QML RowLayout: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
2020-02-16 20:23:54.931	W qrc:/components/SuccessfulTxDialog.qml:109:9: QML ColumnLayout: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
2020-02-16 20:23:54.931	W qrc:/components/SuccessfulTxDialog.qml:167:9: QML RowLayout: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
2020-02-16 20:23:54.931	W qrc:/components/SuccessfulTxDialog.qml:109:9: QML ColumnLayout: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
2020-02-16 20:23:54.931	W qrc:/components/SuccessfulTxDialog.qml:167:9: QML RowLayout: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
```

Comment (rating89us) [reply components/TxConfirmationDialog.qml]: I'm using log level 4 and these warning messages are not being displayed. How do you turn this on?

Comment (selsta) [reply components/TxConfirmationDialog.qml]: Which Qt version are you using? 5.9 might not display the warning. You can install newer Qt versions using the installer.

**Comment by selsta** [comment]
> Trezor doesn't return transaction fee (issue #2768)

I think this should be a separate PR.

**Comment by rating89us** [comment]
> > Trezor doesn't return transaction fee (issue #2768)
> 
> I think this should be a separate PR.

Removed from to do list. The problem is that the final confirmation that is displayed to the user (after transaction was confirmed in Trezor device) still displays "Fee: See on device", but the device is not displaying the fee anymore.

**Comment by selsta** [comment]
> - Destination address of pending (0 confirmation) transactions is now being displayed (fixes #2008)
> - Search filter now works when there is a pending transaction (fixes #2703)
> - Auto collapse when there is a single result

Please move all of these to separate PRs, I think they not related to this one? This makes it easier to review things.

**Comment by rating89us** [comment]
> > * Destination address of pending (0 confirmation) transactions is now being displayed (fixes #2008)
> > * Search filter now works when there is a pending transaction (fixes #2703)
> > * Auto collapse when there is a single result
> 
> Please move all of these to separate PRs, I think they not related to this one? This makes it easier to review things.

#2703 is required for the `View progress` button to work. 
#2008 and "Auto collapse when there is a single result" can be moved to other PRs.

**Comment by selsta** [comment]
Okay, sounds good.

**Comment by selsta** [comment]
btw #2739 is merged now

**Comment by rating89us** [comment]
This PR will be split in two PRs: please see PR #2895

**Pull request closed without merging**